amundsen_application/static/js/ducks/search/reducer.ts

```diff
@@ -104,6 +104,7 @@ export default function reducer(state: SearchReducerState = initialState, action
     case SearchResource.FAILURE:
       return {
         ...initialState,
+        search_term: (<SearchAllResponse>action).payload.search_term,
```

Comment:
This does make sense, but let me clarify what I was trying to get at:
1. This current executes for `SearchAll.FAILURE` and `SearchResource.FAILURE`. However you haven't modified the `searhResearchWorker` saga to return a payload for `SearchResource.FAILURE`. It will be a better practice to separate the two cases regardless, as right now we've introduced a side effect for `SearchResource.FAILURE` as well. If `SearchResource` actually does fail, our app will current break because `payload` will be undefined. Let's fix that before we merge.
2. I referenced #201 because it makes some changes that has to modify this code regardless, so you had the option to not do throw-away work if you wanted. 
The issue that will have to be fixed/re-written regardless of what we merge in here has to do with this line in the sagas
```
yield put({ type: SearchAll.FAILURE, payload: {search_term: term} });
```
 Notice that our current code did not complain about the fact that you created a payload that only had `search_term` even though the `SearchAllResponse` type requires more on the payload according to the type we specified: https://github.com/lyft/amundsenfrontendlibrary/blob/master/amundsen_application/static/js/ducks/search/types.ts#L34. We're missing some type safety here and #201 cleans that up.

Reply:
ok, so in that case I should just roll back the changes related to the faulty query?

Reply:
yeah feel free

Reply:
:+1: